powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/generator/IdentifierGenerator.java

```diff
@@ -56,6 +56,11 @@ public class IdentifierGenerator {
      */
     private static final int ACTIVATION_CODE_BYTES_LENGTH = 12;
 
+    /**
+     * When {@code ACTIVATION_CODE_BYTES_LENGTH = 12}, the Base32 padding is always {@code ====}.
+     */
+    private static final String ACTIVATION_CODE_PADDING = "====";
```

Comment:
What happens in case `ACTIVATION_CODE_BYTES_LENGTH` constant would change? Isn't it better to calculate the padding characters?

Reply:
🤔 I am not merging it yet. I will return to this later. The thing is that BC decoder is much more strict and unlike Guva it throws an exception if it is without padding. So we transform for example
`AAAAAAAAAAAAAAAAAAAA====` into `AAAAA-AAAAA-AAAAA-AAAAA` but decoding of `AAAAAAAAAAAAAAAAAAAA` would fail.

Reply:
See: https://github.com/bcgit/bc-java/blob/4ee2ee57b2ed6e8a80120c5c90796438701e2b88/core/src/main/java/org/bouncycastle/util/encoders/Base32Encoder.java#L90

Reply:
Thanks for the hint. Now the padding is not fixed anymore but calculated.